src/libsyntax/parse/parser.rs

```diff
@@ -5781,12 +5781,19 @@ impl<'a> Parser<'a> {
             }
             token::CloseDelim(token::Brace) => {}
             token::DocComment(_) => {
-                let mut err = self.span_fatal_err(self.span, Error::UselessDocComment);
+                let current_span = self.span;
+                let current_token = self.this_token_to_string();
+                let mut err = self.span_fatal_err(current_span, Error::UselessDocComment);
+
                 self.bump(); // consume the doc comment
                 if self.eat(&token::Comma) || self.token == token::CloseDelim(token::Brace) {
                     err.emit();
                 } else {
-                    return Err(err);
+                    let mut comma_err = self.span_fatal_help(current_span,
+                                    &format!("expected `,`, or `}}`, found `{}`", current_token),
+                                    "struct fields should be separated by commas");
+                    err.emit();
+                    return Err(comma_err);
```

Comment:
I'm not convinced that this code will perform the expected behavior. `current_token` would, if I'm reading this correctly, be the doc comment.

I think that what you'd want to do is keep a `mut seen_comma: bool` that gets set to true either if `self.token` is `Comma` or if the comma is `eat`en after consuming the doc comment. If that happens, _then_  you perform the logic in the `else` branch above. Makes sense?

Reply:
@estebank : If we are inside `token::DocComment(_)` block instead of `token::Comma` then it means current token is not `comma` but `doc`, from here we can imply that `comma` is missing between `doc comment` and `struct field` and if next token is not `comma` then we are going inside `else` block, this is where, we are emitting `comma missing message`. This is my current approach, I will go with your approach because that looks much cleaner version :)

Now problem is my testcase is failing because of unknown reasons. Problem is test case will fail even if we write testcase for `default` block of `switch case`. Any tips how I can fix my testcase? 

Let me know if I am getting all this wrong :)

Reply:
The test is failing due to missing whitespace in the missing main error:

```
[00:43:14] 
[00:43:14] ---- [ui] ui/issue-48636.rs stdout ----
[00:43:14] 	diff of stderr:
[00:43:14] 
[00:43:14] 15	   = help: struct fields should be separated by commas
[00:43:14] 16	
[00:43:14] 17	error[E0601]: `main` function not found in crate `issue_48636`
[00:43:14] -	  |
[00:43:14] -	  = note: consider adding a `main` function to `$DIR/issue-48636.rs`
[00:43:14] +	   |
[00:43:14] +	   = note: consider adding a `main` function to `$DIR/issue-48636.rs`
[00:43:14] 20	
[00:43:14] 21	error: aborting due to 3 previous errors
[00:43:14] 22	
```

Add an empty `fn main() {}` in the file so that we reduce the amount of text in the `stderr` file only to the relevant diagnostic. Also, you'll have to annotate the `rs` test file with a comment in line 16 as follows:

```
//~^^^ ERROR found a documentation comment that doesn't document anything
```

Each `^` makes the check happen one line up, and then it just matches the text against the diagnostics emitted.